File: lib-util/src/main/java/code/util/TrueHostnameVerifier.java
package code.util;

import android.text.TextUtils;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.HttpsURLConnection;
import javax.net.ssl.SSLSession;

public class TrueHostnameVerifier implements HostnameVerifier {
    public  String domain;

    public TrueHostnameVerifier(String domain) {
        this.domain = domain;
    }
    public TrueHostnameVerifier() {
    }

    @Override
    public boolean verify(String hostname, SSLSession session) {
        if(TextUtils.isEmpty(domain)) {
            domain = hostname;
        }
        return HttpsURLConnection.getDefaultHostnameVerifier().verify(domain, session);
    }
}

File: lib-util/src/main/java/code/util/TrueHostnameVerifier.kt
package code.util

import android.text.TextUtils
import javax.net.ssl.HostnameVerifier
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLSession

class TrueHostnameVerifier : HostnameVerifier {
    var domain: String? = null

    constructor(domain: String?) {
        this.domain = domain
    }

    constructor()

    override fun verify(hostname: String, session: SSLSession): Boolean {
        if (TextUtils.isEmpty(domain)) {
            domain = hostname
        }
        return HttpsURLConnection.getDefaultHostnameVerifier().verify(domain, session)
    }
}